python/pip_install/extract_wheels/extract_single_wheel.py

```diff
@@ -8,11 +8,32 @@
 
 from python.pip_install.extract_wheels import arguments, bazel, requirements
 from python.pip_install.extract_wheels.annotation import annotation_from_str_path
-from python.pip_install.extract_wheels.extract_wheels import (
-    configure_reproducible_wheels,
-)
 
 
+def configure_reproducible_wheels() -> None:
+    """Modifies the environment to make wheel building reproducible.
+    Wheels created from sdists are not reproducible by default. We can however workaround this by
+    patching in some configuration with environment variables.
+    """
+
+    # wheel, by default, enables debug symbols in GCC. This incidentally captures the build path in the .so file
+    # We can override this behavior by disabling debug symbols entirely.
+    # https://github.com/pypa/pip/issues/6505
+    if "CFLAGS" in os.environ:
```

Comment:
We should be careful with environment variables for the C and C++ compilers. Ideally, we should instruct users to set these instead of doing them ourselves. Setting `-g0` is only part of making them deterministic.

If we determine that we actually want to mess with them, then setting `CPPFLAGS` is where we want to start.

Reply:
I agree. I'm not really confident in these either. My personal preference would be to not have any sdist being built in repository rules at all and encourage "binary-only" wheel installations instead.

All I've done here is to bring over the existing function that was already being called in both `pip_install` and `pip_parse` that was previously in extract_wheels.py (which will be deleted in this PR).

Reply:
It's good to keep it this way for now. I thought I had commented on this already. We can make improvements to this part on a follow-up PR.